src/Hazel.hs

```diff
@@ -367,40 +367,46 @@ check dirs ctx ty tm = case tm of
         "[check Let] must consume linear bound variables"
     return leftovers2
 
-  Tuple modality vTms -> case ty of
-    -- Thread the leftover context through from left to right.
-    TimesTy tys -> do
-      assert (modality == LinearUnpack) dirs
-        "[check Tuple] tuple modality must be linear unpack for Times"
-
-      -- Layer on a state transformer for this bit, since we're passing
-      -- leftovers from one term to the next
-      let calc = V.imapM
-            (\i (tm', ty') -> do
-              let dirs' = (Tuple' i):dirs
-              leftovers <- get
-              newLeftovers <- lift $ check dirs' leftovers ty' tm'
-              put newLeftovers
-            )
-            (V.zip vTms tys)
-
-      -- execState gives back the final state
-      execStateT calc ctx
-
-
-    -- make sure each of the branches checks and consumes the same resources
-    WithTy tys -> do
-      assert (modality == LinearProject) dirs
-        "[check Tuple] tuple modality must be linear projection for With"
-
-      allLeftovers <- flip V.imapM vTms $ \i val -> do
-        check (Tuple' i:dirs) ctx (tys V.! i) val
-      assert (allTheSame (V.toList allLeftovers)) dirs
-        "[check WithTy] mismatching leftovers"
-      return (V.head allLeftovers)
+  Tuple modality vTms -> do
+    let threadedLeftovers tms tys =
```

Comment:
I think it's worth a note that we don't use this in the `With` case -- only `Times` and `Tuple`.